core/msd/src/main/rdl/KubernetesNetworkPolicy.tdl

```diff
@@ -0,0 +1,74 @@
+type KubernetesLabelSelector Struct {
+    Array<KubernetesLabelSelectorRequirement> matchExpressions; //Array of label selector requirements. The requirements are ANDed.
+    Map<String, String> matchLabels; //Map of label key/value pairs
+}
+
+// Kubernetes network policy port range
+type KubernetesNetworkPolicyPort PolicyPort {
+    TransportPolicyProtocol protocol; //Network policy protocol. Allowed values: TCP, UDP.
+}
+
+// Kubernetes network policy IP block source/target
+type KubernetesIPBlock Struct {
+    String cidr; //CIDR block representing IP range for source/target
+    Array<String> except (optional); //Exception for CIDR blocks, if needed
+}
+
+// Kubernetes network policy peer (source/target)
+type KubernetesNetworkPolicyPeer Struct {
+    KubernetesLabelSelector podSelector (optional); //Kubernetes pod selector for the network policy source/target
+    KubernetesLabelSelector namespaceSelector (optional); //Kubernetes namespace selector for the network policy source/target
+    KubernetesIPBlock ipBlock (optional); //IP block for the network policy source/target
+}
+
+// Kubernetes network policy ingress rule
+type KubernetesNetworkPolicyIngressRule Struct {
+    Array<KubernetesNetworkPolicyPeer> from (optional); //Network policy source, when empty all sources are allowed
+    Array<KubernetesNetworkPolicyPort> ports (optional); //Ingress port(s), when empty all ports are allowed
+}
+
+// Kubernetes network policy egress rule
+type KubernetesNetworkPolicyEgressRule Struct {
+    Array<KubernetesNetworkPolicyPeer> to (optional); //Network policy target, when empty all sources are allowed
+    Array<KubernetesNetworkPolicyPort> ports (optional); //Egress port(s), when empty all ports are allowed
+}
+
+// Kubernetes network policy spec
+type KubernetesNetworkPolicySpec Struct {
+    KubernetesLabelSelector podSelector; //Kubernetes pod selector for the network policy target
+    Array<String> policyTypes; //Network policy types - Ingress, Egress
+    Array<KubernetesNetworkPolicyIngressRule> ingress (optional); //Ingress network policy rules, if empty then all ingress traffic is blocked
+    Array<KubernetesNetworkPolicyEgressRule> egress (optional); //Egress network policy rules, if empty then all egress traffic is blocked
+}
+
+// Request object containing Kubernetes network policy inputs
+type KubernetesNetworkPolicyRequest Struct {
+    String athenzDomainLabel (optional); //Label key name used on pods to identify Athenz domain
+    String athenzServiceLabel; //Label key name used on pods to identify Athenz service
+    String networkPolicyType (optional); //Network policy type, default is vanilla Kubernetes
+    String requestedApiVersion (optional); //Requested network policy apiVersion
+    String networkPolicyNamespace (optional); //Kubernetes namespace for the network policy target
+    KubernetesLabelSelector podSelector (optional); //Kubernetes pod selector for the network policy target
+}
```

Comment:
I think one option is missing - whether to use a combination of namespace selector and label selector to form the k8s policies OR use only the label selector.

Reply:
Both the fields are optional for that reason. Caller can set either or both or none.

Reply:
upon further review, noticed this definition uses namespace selector as well, so let me rephrase the previous question - how can user choose between `combination of namespace selector and label selector` and `only the label selector, using athenzServiceLabel` ?